Corona-Warn-App/src/main/java/de/rki/coronawarnapp/eventregistration/events/ui/category/TraceLocationCategoryFragment.kt

```diff
@@ -0,0 +1,48 @@
+package de.rki.coronawarnapp.eventregistration.events.ui.category
+
+import android.os.Bundle
+import android.view.View
+import android.view.accessibility.AccessibilityEvent
+import androidx.fragment.app.Fragment
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.databinding.EventRegistrationCategoryFragmentBinding
+import de.rki.coronawarnapp.eventregistration.events.ui.category.adapter.CategoryItem
+import de.rki.coronawarnapp.eventregistration.events.ui.category.adapter.TraceLocationCategoryAdapter
+import de.rki.coronawarnapp.eventregistration.events.ui.category.adapter.category.TraceLocationUIType
+import de.rki.coronawarnapp.eventregistration.events.ui.category.adapter.category.traceLocationCategories
+import de.rki.coronawarnapp.eventregistration.events.ui.category.adapter.header.TraceLocationHeaderItem
+import de.rki.coronawarnapp.eventregistration.events.ui.category.adapter.separator.TraceLocationSeparatorItem
+import de.rki.coronawarnapp.util.ui.popBackStack
+import de.rki.coronawarnapp.util.ui.viewBindingLazy
+import timber.log.Timber
+
+class TraceLocationCategoryFragment : Fragment(R.layout.event_registration_category_fragment) {
+
+    private val binding: EventRegistrationCategoryFragmentBinding by viewBindingLazy()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        binding.toolbar.setNavigationOnClickListener { popBackStack() }
+
+        val categoryList = mutableListOf<CategoryItem>().apply {
+            add(TraceLocationHeaderItem(R.string.tracelocation_organizer_category_type_location_header))
+            addAll(traceLocationCategories.filter { it.uiType == TraceLocationUIType.LOCATION })
+            add(TraceLocationSeparatorItem)
+            add(TraceLocationHeaderItem(R.string.tracelocation_organizer_category_type_event_header))
+            addAll(traceLocationCategories.filter { it.uiType == TraceLocationUIType.EVENT })
+        }
```

Comment:
Could you please move it to a ViewModel ? it will be easier for testing and mocking the data in the screen if required what do you think?

Reply:
Yes, good idea! Moved it to a ViewModel and also added a test for it.